rest/route/project_copy.go

```diff
@@ -91,3 +91,83 @@ func (p *projectCopyHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(apiProjectRef)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/projects/{project_id}/copy_redacted
+
+type copyRedactedVarsHandler struct {
+	copyFrom string
+	copyTo   string
+	dryRun   bool
+	sc       data.Connector
+}
+
+func makeCopyRedactedVars(sc data.Connector) gimlet.RouteHandler {
+	return &copyRedactedVarsHandler{
+		sc: sc,
+	}
+}
+
+func (p *copyRedactedVarsHandler) Factory() gimlet.RouteHandler {
+	return &copyRedactedVarsHandler{
+		sc: p.sc,
+	}
+}
+
+func (p *copyRedactedVarsHandler) Parse(ctx context.Context, r *http.Request) error {
+	p.copyFrom = gimlet.GetVars(r)["project_id"]
+	p.copyTo = r.FormValue("dest_project")
+	p.dryRun = r.FormValue("dry_run") == "true"
+	if p.copyTo == "" {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    "must provide new project ID",
+		}
+	}
+	return nil
+}
+
+func (p *copyRedactedVarsHandler) Run(ctx context.Context) gimlet.Responder {
+	_, err := p.sc.FindProjectById(p.copyTo) // ensure project is existing
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error finding project '%s'", p.copyTo))
+	}
+
+	varsToCopy, err := p.sc.FindProjectVarsById(p.copyFrom, true)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error finding variables for '%s'", p.copyFrom))
+	}
+
+	// return any variables that will be overwritten in the new project
+	if p.dryRun {
+		if err := p.willOverwriteVars(varsToCopy); err != nil {
+			return gimlet.NewJSONErrorResponse(err)
+		}
+		return gimlet.NewJSONResponse(struct{}{})
+	}
+
+	if err := p.sc.UpdateProjectVars(p.copyTo, varsToCopy); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "error copying project vars from project '%s'", p.copyFrom))
+	}
+
+	return gimlet.NewJSONResponse(struct{}{})
+}
+
+func (p *copyRedactedVarsHandler) willOverwriteVars(varsToCopy *model.APIProjectVars) error {
+
+	overwrittenVars := []string{}
+	existingVars, err := p.sc.FindProjectVarsById(p.copyTo, false)
```

Comment:
I don't follow why the argument to this function is false. I understand that it's true on https://github.com/evergreen-ci/evergreen/pull/3085/files#diff-98bc657252c6f51456d1a712693f2f26R137, because we're copying the redacted variables. Here though we're logging the unredacted variables in the too project. Why?

Reply:
I can add a comment: In copyFrom we may have a variable `"apple": "green"` that is redacted, whereas in copyTo we have `"apple": "red"` and it's not redacted. When we update, we will still be overwriting the variable `apple` (and now it will also be redacted), so we want to notify the user about apple.

Reply:
TL;DR we don't just overwrite redacted variables.

Reply:
If I'm copying variables, and I'm about to overwrite a redacted and an unredacted variable, does this line catch it? I think it only returns redacted or not redacted right?

Reply:
We're only copying redacted variables: it catches if it overwrites any variable in the new project (redacted or unredacted). This check is only done with --dryRun also, since users might not care / may want that variable overwritten.

Reply:
(existingVars.Vars contains both private and public variable keys for the destination project)